src/languageclient.rs

```diff
@@ -2202,16 +2202,42 @@ impl State {
         let user_data: VimCompleteItemUserData = serde_json::from_str(&user_data)?;
 
         let mut edits = vec![];
+        let mut cur_line = line + 1;
+        let mut cur_char = character + 1;
```

Comment:
Here I'm setting up some mutable variables for the line/char index.  This can change due to additional text inserts, etc.

Reply:
This and following calculation are all 0-based. It might be better to keep 0-based until calling vim api.